contracts/upgradeable_contracts/BaseRelativeDailyLimit.sol

```diff
@@ -0,0 +1,85 @@
+pragma solidity 0.4.24;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "./BasicTokenBridge.sol";
+
+contract BaseRelativeDailyLimit is BasicTokenBridge {
+    using SafeMath for uint256;
+
+    event TargetLimitChanged(uint256 newLimit);
+    event ThresholdChanged(uint256 newThreshold);
+    event TodayLimitSet(uint256 limit);
+
+    bytes32 internal constant TARGET_LIMIT = 0x192ac2d88a9de45ce541663ebe1aaf6d6b1d4a6299d3fd0abf2ba7e8b920342b; // keccak256(abi.encodePacked("targetLimit"))
+    bytes32 internal constant THRESHOLD = 0xd46c2b20c7303c2e50535d224276492e8a1eda2a3d7398e0bea254640c1154e7; // keccak256(abi.encodePacked("threshold"))
+
+    function _calculateLimit() internal view returns (uint256) {
+        uint256 balance = _getTokenBalance();
+        uint256 unlimitedBalance = _minPerTx();
+        if (balance <= unlimitedBalance) {
+            return balance;
+        }
+        uint256 limit = targetLimit();
+        uint256 thresh = threshold();
+        uint256 multiplier = 1 ether ** 2;
+        if (balance > unlimitedBalance && balance < thresh) {
```

Comment:
I think the check `balance > unlimitedBalance` is not needed since it will always be `true` because of previous check 
```javascript
 if (balance <= unlimitedBalance) {
    return balance;
}
```

Reply:
Yes, you are right. I saw it, but decided not to change change the proposed implementation https://github.com/poanetwork/tokenbridge-contracts/pull/306#discussion_r334659756 until the review